transport.go

```diff
@@ -53,6 +53,8 @@ func (a *Agent) connect(ctx context.Context, isControlling bool, remoteUfrag, re
 		return nil, err
 	}
 
+	close(a.initializedCh)
```

Comment:
We have three ways to communicate `initialized` now. Would it be possible to move them into one thing?

* `Agent.opened`
* `Agent.haveStarted` and `Agent.muHaveStarted`
*  `Agent. initializedCh`

Sorry lots of tech debt has added up. Happy to help aggressively refactor this.

Reply:
`opened` seems never set and the same check is done using `haveStarted`. It can be simply removed.

As it requires to block until start, I'll move them into chan.